kbase-extension/static/kbase/js/common/spec.js

```diff
@@ -101,11 +101,51 @@ define(['bluebird', 'util/util', 'common/sdk', 'widgets/appWidgets2/validators/r
             return Promise.props(validationMap);
         }
 
+        /**
+         * This validates parameter sets by using a bulk form of validator for each parameter type.
+         * If options are available for the parameter validation, they're expected to apply to all
+         * parameters of that type.
+         * This returns a batch of promises that resolves to a arrays of validations, keyed on
+         * parameter id, matching the order of arrays in the paramValues structure.
+         * NOTE: This expects that paramValues is normalized - each array is present and the same
+         * length - otherwise unexpected behavior may occur.
```

Comment:
are there any checks in place to ensure that `paramValues` is normalised prior to being used by this function?

Reply:
ETA: spotted the checks. 🎉